src/tyr/route_serializer_osrm.cc

```diff
@@ -560,8 +583,15 @@ json::ArrayPtr intersections(const valhalla::DirectionsLeg::Maneuver& maneuver,
       auto lanes = json::array({});
       for (const auto& turn_lane : prev_edge->turn_lanes()) {
         auto lane = json::map({});
-        // Process 'valid' flag
-        lane->emplace("valid", turn_lane.is_active());
+        // Process 'valid' & 'active' flags
+        bool is_valid = turn_lane.state() == ::valhalla::TurnLane::kValid;
```

Comment:
the logic is reversed here - it can be valid if it is active OR the state is valid
bool is_active = (turn_lane.state() == ::valhalla::TurnLane::kActive);
bool is_valid = is_active || (turn_lane.state() == ::valhalla::TurnLane::kValid);

Reply:
I'm not sure I follow. Both implementations will be setting the 2 values exactly the same way. I put the valid check first because a lane cannot be active if its not also valid. Is there some other combination of active/valid that I'm missing currently?

Reply:
@mandeepsandhu there is only one state but multiple booleans to return - an `active` is also `valid` so you have to check `is_active` in that case

Reply:
You are totally right. I don't know why I confused myself thinking this was like a "mask" where both attributes could be true. I'll fix this and add tests for serialized response to verify the behavior.